semgrep-core/src/engine/Test_engine.ml

```diff
@@ -89,12 +92,24 @@ let test_rules ?(ounit_context = false) xs =
              let d, b, ext = Common2.dbe_of_filename file in
              Common2.readdir_to_file_list d @ Common2.readdir_to_link_list d
              |> Common.find_some (fun file2 ->
-                    let _, b2, ext2 = Common2.dbe_of_filename_noext_ok file2 in
-                    (* ugly: jsonnet exclusion below because of some .jsonnet and .yaml
-                     * ambiguities in tests/OTHER/rules *)
-                    if b = b2 && ext <> ext2 && ext2 <> "jsonnet" then
-                      Some (Filename.concat d file2)
-                    else None)
+                    let path2 = Filename.concat d file2 in
+                    (* In order to distinguish YAML test files from rule files,
+                     * YAML test files use the .test.yaml extension.
+                     * THINK: If we used .yml or .test_yaml we would not need
+                     *        to handle this case separately. *)
+                    match
+                      Filename.chop_suffix_opt ~suffix:".test.yaml" file2
```

Comment:
I'm surprised ocamlformat use this indentation. Are you sure you set pre-commit correctly?

Reply:
I think it's correctly setup, it probably does that because otherwise the line would be too long. I can introduce a `let` to split it up.